doctest/DoctestDriver.hs

```diff
@@ -0,0 +1,12 @@
+{-# LANGUAGE CPP #-}
+
+#if MIN_VERSION_GLASGOW_HASKELL(8,4,4,0)
+{-# OPTIONS_GHC -F -pgmF doctest-discover #-}
+#else
+module Main where
+
+import qualified System.IO as IO
+
+main :: IO ()
+main = IO.putStrLn "WARNING: doctest will not run on GHC versions earlier than 8.4.4"
+#endif
```

Comment:
Unfortunately `cabal 3` can't generate environment files for GHC versions earlier than `ghc-8.4.x` so in such cases we use `CPP` to substitute a test driver that does nothing.